packages/utils/number/handleNumericCallback/handleNumericCallback.ts

```diff
@@ -3,25 +3,38 @@ import {
   thousandsDelimiter,
 } from "../thousandsDelimiter/thousandsDelimiter";
 
+type Props = {
+  value: string;
+  callback: (value: string) => void;
+  shouldDelimiter?: boolean;
+  delimiter?: string;
+};
+
 /**
  * This function executes the provided callback using the provided value if the value is an empty string or number.
  *
  * Useful for ensuring numerical input in onChange events.
  *
- * @param {string} value - The input value to be processed by the callback function.
- * @param {(value: string) => void} callback - A callback function receiving the value only if it's numeric or an empty string.
- * @param {boolean} [commas=true] - A boolean flag to include thousands delimiter(comma) in the formatted value. (default: true).
- */
-export function handleNumericCallback(
-  value: string,
-  callback: (value: string) => void,
-  commas: boolean = true
-) {
-  const parsedValue = removeThousandsDelimiter(value);
+ * @param {Props} props - The properties object.
+ * @param {string} props.value - The input value to be processed by the callback function.
+ * @param {(value: string) => void} props.callback - A callback function receiving the value only if it's numeric or an empty string.
+ * @param {boolean} [props.shouldDelimiter=true] - A boolean flag to include thousands delimiter in the formatted value. (default: true)
+ * @param {string} [props.delimiter=","] - The thousands delimiter to use. (default: ",")
+ * **/
+
+export function handleNumericCallback({
```

Comment:
I think `executeCbIfNumeric` better describes the function rather than `handleNumericCallback`.

Reply:
Perhaps, throwing an Error if the value is not an empty string or a number could be helpful to the user?

Reply:
I don't think there's any need to throw an error.
Because this function is similar to an `if statement` that only runs if the input is an empty string or a numeric value.